vlo-web-app/src/main/java/eu/clarin/cmdi/vlo/service/solr/impl/SolrDaoImpl.java

```diff
@@ -58,7 +59,7 @@ protected SolrQuery sanitise(SolrQuery query) {
                 String facetInFilter = filter.split(":")[0]
                         .replaceFirst("^\\{!tag=.*\\}", ""); //strip out tag syntax (may be used for OR queries, see query factory implementation)
                 
-                if (FacetConstants.AVAILABLE_FACETS.contains(facetInFilter)) {
+                if (FacetConstants.getAvailableFacets(VloWicketApplication.get().getFieldNameService()).contains(facetInFilter)) {
```

Comment:
I think that the `FieldNameService` can be passed into this class through the constructor (via its children `AutoCompleteServiceImpl` and `SearchResultsDaoImpl`, both of which are Spring managed beans - see `VloSpringConfig`). This would remove the need for an explicit dependency on the `VloWicketApplication` class.

Reply:
this approach was just a work around since any call of getAvailableFacets would create a new instance of an ImmutableSet. In my view we should store any reduced list of fieldnames in the vloConfig object as you mentioned before